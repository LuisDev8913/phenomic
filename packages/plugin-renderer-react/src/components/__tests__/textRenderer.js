/* eslint-disable react/no-multi-comp */

import textRenderer from "../textRenderer";

test("textRenderer tells you when pass undefined", () => {
  const spy = jest.spyOn(console, "error");
  expect(textRenderer()).toMatchSnapshot();
  expect(spy).toHaveBeenCalled();
});

test("textRenderer works with raw html", () => {
  expect(textRenderer("<h1>Test</h1")).toMatchSnapshot();
});

test("textRenderer works with phenomic json body", () => {
  expect(
    textRenderer({
      t: "p",
      p: { className: "test" },
      c: [
        {
          t: "a",
          p: { href: "http://test" },
          c: "Link"
        },
        " ",
        {
          t: "span",
          c: "text"
        },
        " ",
        {
          t: "strong",
          c: "fat"
        }
      ]
    })
  ).toMatchSnapshot();
});
