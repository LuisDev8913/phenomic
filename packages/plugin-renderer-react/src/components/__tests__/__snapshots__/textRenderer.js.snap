// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`textRenderer tells you when pass undefined 1`] = `""`;

exports[`textRenderer works with phenomic json body 1`] = `"Link text fat"`;

exports[`textRenderer works with raw html 1`] = `"Test"`;
